Build Errors with Cuda  ‘struct cv::cuda::device::ThrustAllocator’ has virtual functions and accessible non-virtual destructor

I'm having a compile issues with the latest master branch of opencv from github. This is my first time building OpenCV from source. I followed the instructions I found here
http://docs.opencv.org/3.0-last-rst/doc/tutorials/introduction/linux_install/linux_install.html

The error I was getting  has something to do with ThrustAllocator which I'm not too sure about. 

 /home/skymeson/opencv/modules/core/include/opencv2/core/cuda/utility.hpp:57:45: error: ‘struct cv::cuda::device::ThrustAllocator’ has virtual functions and accessible non-virtual destructor 

My setup is pretty standard Ubuntu 14.04 setup with latest cuda driver. I'm wondering if there is something in my setup causing this error or if it is an issue with latest branch? Is there a flag I could try to turn this feature off or perhaps a different branch I should be cloning? Any help getting OpenCV to compile on my system would be much appreciated. 

### Information about my system
- OpenCV version:  master branch 
- Host OS: Linux (Ubuntu 14.04)  

`gcc -v`
gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.1) 

`cmake --version`
cmake version 2.8.12.2

`nvcc --version`
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2015 NVIDIA Corporation
Built on Tue_Aug_11_14:27:32_CDT_2015
Cuda compilation tools, release 7.5, V7.5.17



Commands used to build OpenCV
git clone https://github.com/Itseez/opencv.git
cd ~/opencv
mkdir release
cd release
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local ..


Build Errors
[  0%] Building NVCC (Device) object modules/core/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_gpu_mat.cu.o
/home/skymeson/opencv/modules/core/include/opencv2/core/cuda/utility.hpp:57:45: error: ‘struct cv::cuda::device::ThrustAllocator’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
     struct CV_EXPORTS ThrustAllocator
                                             ^
/home/skymeson/opencv/modules/core/src/cuda/gpu_mat.cu:61:7: error: ‘class _GLOBAL__N__54_tmpxft_00004115_00000000_13_gpu_mat_compute_35_cpp1_ii_71482d89::DefaultThrustAllocator’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
     class DefaultThrustAllocator: public cv::cuda::device::ThrustAllocator
       ^
/home/skymeson/opencv/modules/core/src/cuda/gpu_mat.cu:74:14: warning: unused parameter ‘numBytes’ [-Wunused-parameter]
         __device__ __host__ void deallocate(uchar* ptr, size_t numBytes)
              ^
cc1plus: some warnings being treated as errors
CMake Error at cuda_compile_generated_gpu_mat.cu.o.cmake:266 (message):
  Error generating file
  /home/skymeson/opencv/release/modules/core/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_gpu_mat.cu.o


make[2]: *** [modules/core/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_gpu_mat.cu.o] Error 1
make[1]: *** [modules/core/CMakeFiles/opencv_core.dir/all] Error 2

--- a/modules/core/include/opencv2/core/cuda/utility.hpp
+++ b/modules/core/include/opencv2/core/cuda/utility.hpp
@@ -57,7 +57,7 @@ namespace cv { namespace cuda { namespace device
     struct CV_EXPORTS ThrustAllocator
     {
         typedef uchar value_type;
-
+        virtual ~ThrustAllocator();
         virtual __device__ __host__ uchar* allocate(size_t numBytes) = 0;
         virtual __device__ __host__ void deallocate(uchar* ptr, size_t numBytes) = 0;
         static ThrustAllocator& getAllocator();

--- a/modules/core/src/cuda/gpu_mat.cu
+++ b/modules/core/src/cuda/gpu_mat.cu
@@ -56,6 +56,9 @@ using namespace cv;
 using namespace cv::cuda;
 using namespace cv::cudev;
 
+device::ThrustAllocator::~ThrustAllocator()
+{
+}
 namespace
 {
     class DefaultThrustAllocator: public cv::cuda::device::ThrustAllocator
@@ -73,6 +76,7 @@ namespace
         }
         __device__ __host__ void deallocate(uchar* ptr, size_t numBytes)
         {
+            (void)numBytes;
 #ifndef __CUDA_ARCH__
             CV_CUDEV_SAFE_CALL(cudaFree(ptr));
 #endif

